api/python/tests/test_cli.py

```diff
@@ -41,5 +42,38 @@ def test_push(self):
                 main.main(('push', '--dir', tmp_dir, name))
 
                 mocked_package_class.assert_called_once_with(quilt3.Package)
-                mocked_set_dir.assert_called_once_with('.', tmp_dir)
+                mocked_set_dir.assert_called_once_with('.', tmp_dir, meta=None)
                 mocked_push.assert_called_once_with(name, registry=None, dest=None, message=None)
+
+
+@pytest.mark.parametrize(
+    'arg, meta, expected_set_dir_count, expected_push_count, expected_meta',
+    [
+        (None, None, 1, 1, None),
+        ('--meta', '{invalid: meta}', 0, 0, {}),
```

Comment:
this should xfail with QuiltException?

Reply:
It doesn't fail since QuiltException is already handled in [main.py](https://github.com/quiltdata/quilt/blob/aed61e374043355d15c9aaa3e01e9a9418fb703a/api/python/quilt3/main.py#L439)

Reply:
sure we should still check that the exception is thrown. one way is patch error handler and make sure it gets called.